src/hott/05-sigma.rzk.md

```diff
@@ -428,3 +428,23 @@ Products distribute inside Sigma types:
       ( ( \ ((a , b) , c) -> (a , (b , c)) , \ _ -> refl) ,
         ( \ ((a , b) , c) -> (a , (b , c)) , \ _ -> refl)))
 ```
+
+## Currying
+
+This is the dependent version of the currying equivalence.
+
+```rzk
+#def equiv-dependent-curry
+  (A : U)
+  (B : A -> U)
+  (C : (a : A) -> B a -> U)
+  : Equiv
+      ((p : Σ (a : A), B a) -> C (first p) (second p))
```

Comment:
Seems like this is a good case for allowing patterns in arguments of Pi types. Would you prefer the following notation here?

```
  : Equiv
      (((a, b) : Σ (a : A), B a) -> C a b)
      ((a : A) -> (b : B a) -> C a b)
```

Reply:
Yes, I think this would be great! Thank you @fizruk!

Reply:
Agreed. If that's something you're able to implement for us that would be great!